### PR TITLE
Feature icetea mactester

### PR DESCRIPTION
**Pull request by JuhPuur**
### Description

Add nanostack MAC tester and testcases to mbed OS.
Stripped out the .lib's, Jenkinsfile and license. RF drivers should hopefully be included in 5.10. Keeping them as .lib's would require having the tester a subrepository.

### Pull request type

- [ ] Fix
- [ ] Refactor
- [ ] New target
- [x] Feature
- [ ] Breaking change

### CODE REVIEW

File: tools/test_configs/config_paths.json [hunk 1]
```diff
@@ -8,5 +8,5 @@
     "ESP8266_WIFI" : "ESP8266Interface.json",
     "WICED_WIFI" : "WicedInterface.json",
     "ISM43362_WIFI" : "ISM43362Interface.json",
-    "IDW0XX1_WIFI" : "SpwfSAInterface.json"
+    "TEST_APP" : "test_app.json"
```
Comment (jarlamsa) [inline tools/test_configs/config_paths.json]: Is this removal of IDW0XX1_WIFI config intentional? Also the "TEST_APP" could be something more informal if it relates only to the mactester.

File: tools/test_configs/target_configs.json [hunk 2]
```diff
@@ -8,12 +8,12 @@
         "test_configurations": ["REALTEK_WIFI"]
     },
     "K64F": {
-        "default_test_configuration": "HEAPBLOCKDEVICE_AND_ETHERNET",
-        "test_configurations": ["HEAPBLOCKDEVICE_AND_ETHERNET", "ESP8266_WIFI", "ETHERNET"]
+        "default_test_configuration": "TEST_APP",
```
Comment (jarlamsa) [inline tools/test_configs/target_configs.json]: Default config should probably be the original

File: tools/test_configs/target_configs.json [hunk 3]
```diff
@@ -8,12 +8,12 @@
...
+        "test_configurations": ["HEAPBLOCKDEVICE_AND_ETHERNET", "TEST_APP", "ESP8266_WIFI", "ETHERNET"]
     },
     "NUCLEO_F429ZI": {
-        "default_test_configuration": "HEAPBLOCKDEVICE_AND_ETHERNET",
-        "test_configurations": ["HEAPBLOCKDEVICE_AND_ETHERNET"]
+        "default_test_configuration": "TEST_APP",
```
Comment (jarlamsa) [inline tools/test_configs/target_configs.json]: Default config should probably be the original

File: tools/test_configs/test_app.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
+{
```
Comment (jarlamsa) [inline tools/test_configs/test_app.json]: Name of the .json should be more informal

**Review by jarlamsa — CHANGES_REQUESTED** [review]

File: tools/test_configs/config_paths.json (hunk 1 above)
Comment (JuhPuur) [reply tools/test_configs/config_paths.json]: Probably by accident. Will rename config.

File: tools/test_configs/config_paths.json [hunk 5]
```diff
@@ -4,9 +4,10 @@
     "HEAPBLOCKDEVICE_AND_ETHERNET": "HeapBlockDeviceAndEthernetInterface.json",
     "ODIN_WIFI" : "OdinInterface.json",
     "ODIN_ETHERNET" : "Odin_EthernetInterface.json",
-    "REALTEK_WIFI" : "RealtekInterface.json",	
+    "REALTEK_WIFI" : "RealtekInterface.json",
```
Comment (jarlamsa) [inline tools/test_configs/config_paths.json]: There's some whitespace characters at the end of line, otherwise looks good to me

**Review by jarlamsa — APPROVED** [review]

**Comment by JuhPuur** [comment]
This is now dependent on:
* https://github.com/ARMmbed/mbed-os/pull/7778
* inclusion of RF drivers to mbed OS https://github.com/ARMmbed/mbed-os/pull/7814

With those requisites fulfilled this now compiles and tests pass

**Pull request merged**